contracts/carbon/Strategies.sol

```diff
@@ -178,14 +178,22 @@ abstract contract Strategies is Initializable {
     // accumulated fees per token
     mapping(Token => uint256) internal _accumulatedFees;
 
+    // mapping between a pair id to its custom trading fee (in units of PPM)
+    mapping(uint128 => uint32) internal _tradingFeePPMOverrides;
+
     // upgrade forward-compatibility storage gap
-    uint256[MAX_GAP - 4] private __gap;
+    uint256[MAX_GAP - 5] private __gap;
 
     /**
      * @dev triggered when the network fee is updated
      */
     event TradingFeePPMUpdated(uint32 prevFeePPM, uint32 newFeePPM);
 
+    /**
+     * @dev triggered when the trading fee override for a given pair is updated
+     */
+    event TradingFeePPMOverridesUpdated(uint128 pairId, uint32 prevFeePPM, uint32 newFeePPM);
```

Comment:
Perhaps `pairId` should be indexed?

Reply:
Or even a pair of tokens instead, since `pairId` is an internal implementation detail.

Reply:
Added indexing